src/internal/connector/onedrive/collections.go

```diff
@@ -622,6 +622,14 @@ func (c *Collections) UpdateCollections(
 			isFolder = item.GetFolder() != nil || item.GetPackage() != nil
 		)
 
+		// TODO(meain): Use `@microsoft.graph.sharedChanged` in
+		// item.GetAdditionalData() to optimize fetching
+		// permissions. When permissions change, this flags is
+		// present, if only data changes, it is not present.  Have to
+		// add `oneDrive.sharedChanged` in `$select` in delta
+		// https://learn.microsoft.com/en-us/onedrive/developer/rest-api
+		// /concepts/scan-guidance#scanning-permissions-hierarchies
+
```

Comment:
is this TODO stale?

Reply:
Nope, this is still pending. I need to do a few tests to be certain of the behaviour. Will add it as a followup later.